### ISSUE TITLE
API key required

### ISSUE THREAD

**Issue by stevecohenfr** [prog-it/yourls-thumbnail-url#2]
It seems that site-shot.com now requesting an api key :

    {"message":"No API key found in request"}

**Comment by stevecohenfr** [comment]
#3 Replace site-shot.com by www.googleapis.com/pagespeedonline

**Comment by ozh** [comment]
@stevecohenfr you should submit a PR to fix this plugin. Alternatively, I'll add your fork to the list of plugins

**Comment by stevecohenfr** [comment]
> @stevecohenfr you should submit a PR to fix this plugin. Alternatively, I'll add your fork to the list of plugins

Yes I did it : https://github.com/prog-it/yourls-thumbnail-url/pull/3

Do I forgot something ?

**Comment by ozh** [comment]
Sorry, I didn't even check before suggesting :)

**Comment by olpo24** [comment]
@stevecohenfr 
Does it even works for you? 
I copied your code from commit an replaced the code from plugin.php.

If I go now to exaple.com/keyword.i I get only a blank screen. 
In Chrome console I get nothing to debug also.

Kind regards,
Ole

**Comment by stevecohenfr** [comment]
> @stevecohenfr
> Does it even works for you?
> I copied your code from commit an replaced the code from plugin.php.
> 
> If I go now to exaple.com/keyword.i I get only a blank screen.
> In Chrome console I get nothing to debug also.
> 
> Kind regards,
> Ole

I updated my PR. Please use my fork : https://github.com/stevecohenfr/yourls-thumbnail-url instead of this repo that seems abandoned. I will update the plugin when it stops to work.

**Comment by olpo24** [comment]
@stevecohenfr
Thank you very much, it works.

I don't know if you want to improve this plugin, but one suggestion:
Perhaps you can implement an an loading circle to pass the time the screenshot service  needs to generate the screenshot?
Kind rergards,
Ole

**Comment by Bogey** [comment]
Steve, it stopped working...
Perhaps we could modify a field in the plugin.php where we can add out API?

**Comment by stevecohenfr** [comment]
> Steve, it stopped working...
> Perhaps we could modify a field in the plugin.php where we can add out API?

Please use my fork (see messages above)

**Comment by Bogey** [comment]
Yeah I did, but could open an issue on the fork repository :-)

**Comment by prog-it** [comment]
Hello to all. I’ll update the plugin soon.
You can use the service https://www.s-shot.ru/. But there is a limit on the number of screenshots created in free mode.

**Comment by superantim** [comment]
> Hello to all. I’ll update the plugin soon. You can use the service https://www.s-shot.ru/. But there is a limit on the number of screenshots created in free mode.

how about using it https://github.com/oscarotero/Embed
can add this library to get thumbnail?

**Comment by superantim** [comment]
> > Hello to all. I’ll update the plugin soon. You can use the service https://www.s-shot.ru/. But there is a limit on the number of screenshots created in free mode.
> 
> how about using it https://github.com/oscarotero/Embed can add this library to get thumbnail?

it worked!
![image](https://user-images.githubusercontent.com/45330570/208059006-7fe784c8-6752-4af8-92ff-34738422fd6a.png)

**Comment by EDIflyer** [comment]
That looks promising @superantim - did you have any luck integrating it?

@stevecohenfr I tried your version but just got an error:
```
Fatal error: Uncaught Error: Call to undefined function imagecreatefromstring() in /var/www/html/user/plugins/thumbnail-url/plugin.php:39 Stack trace: #0 /var/www/html/includes/functions-plugins.php(224): progit_yourls_thumbnail(Array) #1 /var/www/html/includes/functions-plugins.php(275): yourls_apply_filter('loader_failed', Array, true) #2 /var/www/html/yourls-loader.php(67): yourls_do_action('loader_failed', 'bbctech.i') #3 {main} thrown in /var/www/html/user/plugins/thumbnail-url/plugin.php on line 39
```
